src/carousel/carousel.ts

```diff
@@ -163,20 +201,29 @@ export class NgbCarousel implements AfterContentChecked,
     this.wrap = config.wrap;
     this.keyboard = config.keyboard;
     this.pauseOnHover = config.pauseOnHover;
+    this.pauseOnFocus = config.pauseOnFocus;
     this.showNavigationArrows = config.showNavigationArrows;
     this.showNavigationIndicators = config.showNavigationIndicators;
   }
 
-  @HostListener('mouseenter')
-  mouseEnter() {
-    this._mouseHover$.next(true);
+  arrowLeft() {
+    this.prev(NgbSlideEventSource.ARROW_LEFT);
+    this._focusIndicator();
   }
 
-  @HostListener('mouseleave')
-  mouseLeave() {
-    this._mouseHover$.next(false);
+  arrowRight() {
+    this.next(NgbSlideEventSource.ARROW_RIGHT);
+    this._focusIndicator();
   }
 
+  set mouseHover(value: boolean) { this._mouseHover$.next(value); }
+
+  get mouseHover() { return this._mouseHover$.value; }
+
+  set focused(value: boolean) { this._focused$.next(value); }
+
+  get focused() { return this._focused$.value; }
+
```

Comment:
It might sound nitpicking here, but could you please move these getters/setters above the constructor ?

Reply:
I have added a commit with this change.